crates/sui-core/src/authority/authority_store.rs

```diff
@@ -88,6 +88,12 @@ pub struct SuiDataStore<const ALL_OBJ_VER: bool, S> {
     /// the same response for any call after the first (ie. make certificate processing idempotent).
     effects: DBMap<TransactionDigest, TransactionEffectsEnvelope<S>>,
 
+    /// This is a map between the transaction digest and its timestamp (UTC timestamp in
+    /// **milliseconds** since epoch 1/1/1970). A transaction digest is subjectively time stamped
+    /// on a node according to the local machine time, so it varies across nodes.
+    /// The timestamping happens when the node sees a txn certificate for the first time.
+    timestamps: DBMap<TransactionDigest, u64>,
```

Comment:
i have a mild preference for moving this to IndexStore (possibly renaming that store if necessary). I'd like to keep SuiDataStore from getting any bigger, and eventually it should hold only safety-critical sui state.

Reply:
that's actually not a bad idea. Timestamp is interesting because I consider its criticalness is between the must-have ones (certificates & effects) and nice-to-have ones (indices). I will change as you suggest